spec/acceptance/sensu_bonsai_asset.rb

```diff
@@ -15,6 +15,15 @@
         version  => '0.1.0',
         provider => 'sensu_api',
       }
+      sensu_bonsai_asset { 'sensu/sensu-ruby-runtime in default':
+        ensure  => 'present',
+        version => 'latest',
+      }
+      sensu_namespace { 'dev': ensure => 'present' }
+      sensu_bonsai_asset { 'sensu/sensu-ruby-runtime in dev':
```

Comment:
does this resource type know to auto require the dev namespace above?

Reply:
Yes.  We have the tests setup for all things with `namespace` property.

Auto requires:

https://github.com/sensu/sensu-puppet/blob/master/lib/puppet/type/sensu_bonsai_asset.rb#L40

Leads to:

https://github.com/sensu/sensu-puppet/blob/master/lib/puppet_x/sensu/type.rb#L29

Spec tests are shared for all things that use a namespace 

https://github.com/sensu/sensu-puppet/blob/master/spec/unit/sensu_bonsai_asset_spec.rb#L232

Leads to:

https://github.com/sensu/sensu-puppet/blob/master/spec/shared_examples/types.rb#L79